File: example/arl.php
<?php

require_once 'Deductions.php';

class arl extends Deductions {
    protected $arlRate;

    public function __construct(float $arlRate = 0.052) {
        parent::__construct(0, 0, $arlRate); // Salud y Pension en 0 para esta clase
        $this->arlRate = $arlRate;
    }

    public function discounts(float $salary): float {
        return $salary * $this->arlRate;
    }
}

?>

File: example/java/arl.java
public class arl extends deductions {
    public static void main(String[] args) {
        // Example usage of the arl class
        arl exampleArl = new arl();
        exampleArl.setArlRate(0.052);
        System.out.println("ARL Rate: " + exampleArl.arlRateLocal);
    }
    private double arlRateLocal;

    public arl(double arlRate) {
        super(0.0, 0.0, 0.0); // Adjusted to match deductions constructor
        this.arlRateLocal = arlRate;
    }
    public arl() {
        this(0.052); // Default ARL rate
    }

    @Override
    public double discounts(double salary) {
        // Example implementation
        return salary * 0.01; // 1% deduction
    }


    public void setArlRate(double arlRate) {
        this.arlRate = arlRate;
    }
}